htlcswitch/link.go

```diff
@@ -3778,6 +3792,50 @@ func (l *channelLink) processRemoteAdds(fwdPkg *channeldb.FwdPkg,
 	l.forwardBatch(replay, switchPackets...)
 }
 
+// experimentalEndorsement returns the value to set for our outgoing
+// experimental endorsement field, and a boolean indicating whether it should
+// be populated on the outgoing htlc.
+func (l *channelLink) experimentalEndorsement(
+	customUpdateAdd record.CustomSet) ([]byte, bool) {
```

Comment:
it seems that the second return value here is redundant. In every case where we have a true relay, we have a non-nil byte. Every case where we have a false relay we have a nil byte.

On another note, it seems like the "correct" return value here would be Option[byte]. We don't need to lift to a byte array here. I'm less attached to making this a non-slice, but I do think it'd be good to ditch the boolean.

Reply:
def, I think I wrote this in a pre-Option (or pre my full understanding of go generics) world 👍